pandas/tests/indexes/multi/test_equivalence.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+
+
+import numpy as np
+import pandas as pd
+import pandas.util.testing as tm
+from pandas import (Index, MultiIndex, PeriodIndex, RangeIndex, Series, compat,
+                    isna)
+from pandas.compat import lrange, lzip, range
+
+
+def test_equals(idx):
+    # TODO: Remove or Refactor. MultiIndex not tested.
+    for name, idx in compat.iteritems({'idx': idx}):
+        assert idx.equals(idx)
+        assert idx.equals(idx.copy())
+        assert idx.equals(idx.astype(object))
+
+        assert not idx.equals(list(idx))
+        assert not idx.equals(np.array(idx))
+
+        # Cannot pass in non-int64 dtype to RangeIndex
+        if not isinstance(idx, RangeIndex):
+            same_values = Index(idx, dtype=object)
+            assert idx.equals(same_values)
+            assert same_values.equals(idx)
+
+        if idx.nlevels == 1:
+            # do not test MultiIndex
+            assert not idx.equals(pd.Series(idx))
+
+
+def test_equals_op(idx):
+    # GH9947, GH10637
+    index_a = idx
+    if isinstance(index_a, PeriodIndex):
```

Comment:
prob can remove this (the isinstance check)

Reply:
Done